sway-features/src/lib.rs

```diff
@@ -0,0 +1,245 @@
+use std::collections::HashMap;
+
+use clap::{Parser, ValueEnum};
+
+macro_rules! features {
+    ($($name:ident = $enabled:literal, $url:literal),* $(,)?) => {
+        paste::paste! {
+            pub const CFG: &[&str] = &[
+                $(
+                    stringify!([<experimental_ $name:snake>]),
+                )*
+            ];
+
+            #[derive(Copy, Clone, Debug, ValueEnum)]
+            #[value(rename_all = "snake")]
+            pub enum Features {
```

Comment:
A small detail that suddenly became obvious when I looked at the new `set_enabled(&mut self, feature: Features, enabled: bool)`. The `feature: Feature*s*` part. `Features` should actually be `Feature` (singular) to follow the Rust convention and the one we currently have (`TypeInfo`, `TyExpression`, ...).

Reply:
Done.